internal/graph/check_test.go

```diff
@@ -996,6 +996,230 @@ condition condition1(param1: string) {
 	require.False(t, resp.Allowed)
 }
 
+func TestCheckDispatchCount(t *testing.T) {
+	ds := memory.New()
+	ctx := storage.ContextWithRelationshipTupleReader(context.Background(), ds)
+
+	t.Run("dispatch_count_tuples_to_usersets", func(t *testing.T) {
+		storeID := ulid.Make().String()
+
+		model := parser.MustTransformDSLToProto(`model
+  schema 1.1
+
+type user
+
+type folder
+  	relations
+		define viewer: [user] or viewer from parent
+		define parent: [folder]
+
+type doc
+	relations
+		define viewer: [user] or viewer from parent
+		define parent: [folder]
+`)
+
+		err := ds.Write(ctx, storeID, nil, []*openfgav1.TupleKey{
+			{
+				User:     "user:jon",
+				Relation: "viewer",
+				Object:   "folder:C",
+			},
+			{
+				User:     "folder:C",
+				Relation: "parent",
+				Object:   "folder:B",
+			},
+			{
+				User:     "folder:B",
+				Relation: "parent",
+				Object:   "folder:A",
+			},
+			{
+				User:     "folder:A",
+				Relation: "parent",
+				Object:   "doc:readme",
+			},
+		})
+		require.NoError(t, err)
+
+		checker := NewLocalChecker()
+
+		typesys, err := typesystem.NewAndValidate(
+			context.Background(),
+			model,
+		)
+		require.NoError(t, err)
+
+		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+
+		resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:              storeID,
+			AuthorizationModelID: model.GetId(),
+			TupleKey:             tuple.NewTupleKey("doc:readme", "viewer", "user:jon"),
+			ResolutionMetadata:   &ResolutionMetadata{Depth: 5},
+		})
+		require.NoError(t, err)
+		require.True(t, resp.Allowed)
+
+		require.Equal(t, uint32(3), resp.GetResolutionMetadata().DispatchCount)
+
+		t.Run("direct_lookup_requires_no_dispatch", func(t *testing.T) {
+			resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
+				StoreID:              storeID,
+				AuthorizationModelID: model.GetId(),
+				TupleKey:             tuple.NewTupleKey("doc:readme", "parent", "folder:A"),
+				ResolutionMetadata:   &ResolutionMetadata{Depth: 5},
+			})
+			require.NoError(t, err)
+			require.True(t, resp.Allowed)
+
+			require.Zero(t, resp.GetResolutionMetadata().DispatchCount)
+		})
+	})
+
+	t.Run("dispatch_count_multiple_direct_userset_lookups", func(t *testing.T) {
+		storeID := ulid.Make().String()
+
+		model := parser.MustTransformDSLToProto(`model
+	  schema 1.1
+	
+	type user
+	
+	type group
+	  relations
+	    define member: [user, group#member]
+
+	type document
+	  relations
+		define viewer: [group#member]
+	`)
+
+		err := ds.Write(ctx, storeID, nil, []*openfgav1.TupleKey{
+			{
```

Comment:
Can we reduce this and the others to the shorthand `tuple.NewTupleKey("group:1", "member", "user:jon")`? We strive to do that elsewhere to cut down on the visual boilerplate.

Reply:
The shorthand leaves ambiguity between user and object. Even as I've been making regular contributions, I still have trouble groking it. 

Can we add a linting rule or enforce this style decision?

Reply:
> Can we add a linting rule or enforce this style decision automatically?
That'd be great, but we'd have to build the linter for it. Up until now we've been operating on keeping each other in check (human lint rules 😄 ).

> The shorthand leaves ambiguity between user and object. Even as I've been making regular contributions, I still have trouble groking it.
That's fair, but if you pass the wrong arguments to a function that's just an unintended side-effect, which the unit under test should keep you in check with anyways. For example, if you have a function that does`fmt.Sprintf("%s/%s", var1, var2)` - if you meant `("%s/%s", var2, var2)` then presumably a test should have caught. Similarly, if you inverse `tuple.NewTupleKey(user, relation, object)` instead of `tuple.NewTupleKey(object, relation, user)` then hopefully the thing you're testing would have accounted for that anyways, and if not then the order probably didn't truly matter.